sdks/java/io/solr/src/test/java/org/apache/beam/sdk/io/solr/SolrIOTest.java

```diff
@@ -155,6 +155,23 @@ public void testRead() throws Exception {
     pipeline.run();
   }
 
+  @Test
+  public void testReadAll() throws Exception {
+    SolrIOTestUtils.insertTestDocuments(SOLR_COLLECTION, NUM_DOCS, solrClient);
+
+    PCollection<SolrDocument> output =
+        pipeline
+            .apply(
+                Create.of(
+                    SolrIO.read()
+                        .withConnectionConfiguration(connectionConfiguration)
+                        .from(SOLR_COLLECTION)
+                        .withBatchSize(101)))
+            .apply(SolrIO.readAll());
+    PAssert.thatSingleton(output.apply("Count", Count.globally())).isEqualTo(NUM_DOCS);
```

Comment:
I believe we can improve this test in the future by reading with multiple `Read`s (the goal of adding `readAll()`) and check the content of read messages (not only count).